### PR TITLE
TST (string dtype): un-xfail string tests specific to object dtype

### PR DESCRIPTION
**Pull request by jbrockmendel**
xref https://github.com/pandas-dev/pandas/issues/54792

### CODE REVIEW

File: pandas/tests/test_algos.py [hunk 1]
```diff
@@ -1684,20 +1682,25 @@ def test_unique_complex_numbers(self, array, expected):
...
         "htable, data",
         [
-            (ht.PyObjectHashTable, [f"foo_{i}" for i in range(1000)]),
-            (ht.StringHashTable, [f"foo_{i}" for i in range(1000)]),
+            (
+                ht.PyObjectHashTable,
+                np.array([f"foo_{i}" for i in range(1000)], dtype=object),
+            ),
+            (
+                ht.StringHashTable,
```
Comment (WillAyd) [inline pandas/tests/test_algos.py:1693]: I thought we added a hashtable for strings that didn't use Python objects. @phofl am I misremembering?

**Comment by jbrockmendel** [comment]
Still active

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks @jbrockmendel

**Comment by jorisvandenbossche** [comment]
Manual backport -> https://github.com/pandas-dev/pandas/pull/60180

**Pull request merged**